### ISSUE TITLE
can you give me the test demo?

### ISSUE THREAD

**Issue by ThomasKKL** [confifu/RepNet-Pytorch#3]
I trained 15 epochs, and the test result is relatively poor. Is there something wrong with my test demo? I hope you can provide me with a test demo. Thank you.

**Comment by confifu** [comment]
The `Model.py` file was old, I have updated it. Some of the trained checkpoints are [here](https://drive.google.com/drive/folders/1uYJQTMR6gRXzFVbCfMREv940iLfk27oK?usp=sharing). The notebook that I am using locally is [here](https://drive.google.com/file/d/1sB6SNneHHe6gMgD748uf8cWtfBM7M_qt/view?usp=sharing). Note that for long videos you need to try at slower frame rates, i.e. dividing the video into multiple 64 frame sections. One 64 frame section can have a max of 32-31 repetitions.

**Comment by ThomasKKL** [comment]
ok，I have a try it, and thank you very much

**Comment by ThomasKKL** [comment]
For the same video, I test your model, the effect is not as good as [the paper](https://arxiv.org/abs/2006.15418) 's model; I have a question, for the period length estimator, in this paper, they use multi class classification objective (softmax cross entry) for optimizing the model. How can we make 64 * 32 dimensional labels?

**Comment by confifu** [comment]
>I test your model, the effect is not as good as the paper 's model

The checkpoints that I linked were not trained exhaustively. There is room for improvement by simply training more.

>How can we make 64 * 32 dimensional labels?

Output a (batchsize, 32, frames) tensor from the model. Get rid of `fc1_3` and transpose the last two dimensions of `y1`.

**Comment by qrsforever** [comment]
I try  both  the checkpoints [tr9.pt and tr9_innt.pt], (https://drive.google.com/drive/folders/1uYJQTMR6gRXzFVbCfMREv940iLfk27oK?usp=sharing), the effect is bad too.

using the test2.mp4 from  testvids for test:

![Screenshot from 2021-06-07 17-45-34](https://user-images.githubusercontent.com/1296332/120996163-ab7df280-c7b8-11eb-8a66-4090de3d5df2.png)

**Comment by confifu** [comment]
Is the repetition in the video symmetric? There are some cases where the model performs badly. This is from the paper.
>Double Counting Errors. We observe that a common fail-
ure mode of our model is that for some actions (e.g. jug-
gling soccer ball), it predicts half the count reported by an-
notators. This happens when the model considers left and
right legs’ motion for counting while people tend to con-
sider the ball’s up/down motion resulting in people double
counting the repetitions. We believe such errors are difficult
to isolate in a class-agnostic manner. But they can be fixed
easily with either labeled data or post-processing methods
if the application is known.

Also the OBO error is 0.30 which means that for 30% of videos the model predicted counts that were more than one off from true value.

**Comment by chennankuan** [comment]
> I try both the checkpoints [tr9.pt and tr9_innt.pt], (https://drive.google.com/drive/folders/1uYJQTMR6gRXzFVbCfMREv940iLfk27oK?usp=sharing), the effect is bad too.
> 
> using the test2.mp4 from testvids for test:
> 
> ![Screenshot from 2021-06-07 17-45-34](https://user-images.githubusercontent.com/1296332/120996163-ab7df280-c7b8-11eb-8a66-4090de3d5df2.png)



> I trained 15 epochs, and the test result is relatively poor. Is there something wrong with my test demo? I hope you can provide me with a test demo. Thank you.

do you have the train dataset，If so, I was wondering if you could kindly share it with me?

**Comment by jokingww** [comment]
Hello, it's seems that the checkpoint link is lapsed. Can you please update the checkpoint link?

**Comment by Cecilia-yu412** [comment]
Hello, it's seems that the checkpoint link is lapsed. Can you please update the checkpoint link and train dataset?
